src/codegen/generate.ts

```diff
@@ -695,7 +695,19 @@ export default class ApiGenerator {
       return this.getTypeFromEnum([schema.const]);
     }
     if (schema.type) {
-      // string, boolean, null, number
+      // string, boolean, null, number, array
+      if (Array.isArray(schema.type)) {
+        return factory.createUnionTypeNode(
+          schema.type.map((type) => {
+            if (type === null) return cg.keywordType.null;
+            if (type === "integer") return cg.keywordType.number;
+            if (type in cg.keywordType)
+              return cg.keywordType[type as keyof typeof cg.keywordType];
```

Comment:
Typecasting here because `schema` is still only of type `OpenAPIV3.SchemaObject`. Making that a union with `OpenAPIV3_1.SchemaObject` would be the cleaner way to go, but I don't know how much that entails.